File: scripts/remove-bg.js
import sharp from 'sharp';
import fs from 'fs';

const inputUrl = 'https://hebbkx1anhila5yf.public.blob.vercel-storage.com/WhatsApp%20Image%202026-02-23%20at%2011.49.37%20AM-n785bnHNrlya3FiumHVXY1MHKUYxJd.jpeg';

// Download the image
const response = await fetch(inputUrl);
const buffer = Buffer.from(await response.arrayBuffer());

// Get raw pixel data
const image = sharp(buffer);
const { width, height } = await image.metadata();
const rawData = await image.ensureAlpha().raw().toBuffer();

// Process pixels: make light gray background transparent
const threshold = 210;
const nearThreshold = 190;

for (let i = 0; i < rawData.length; i += 4) {
  const r = rawData[i];
  const g = rawData[i + 1];
  const b = rawData[i + 2];

  if (r > threshold && g > threshold && b > threshold) {
    rawData[i + 3] = 0;
  } else if (r > nearThreshold && g > nearThreshold && b > nearThreshold) {
    const brightness = (r + g + b) / 3;
    const alpha = Math.round(((threshold - brightness) / (threshold - nearThreshold)) * 255);
    rawData[i + 3] = Math.max(0, Math.min(255, alpha));
  }
}

// Save as PNG buffer with transparency, trimmed
const outputBuffer = await sharp(rawData, { raw: { width, height, channels: 4 } })
  .trim()
  .png()
  .toBuffer();

// Write using fs to ensure it goes to the right place
fs.writeFileSync('/vercel/share/v0-project/public/images/logo.png', outputBuffer);
console.log(`Logo saved successfully. Size: ${outputBuffer.length} bytes`);

File: scripts/remove-bg.py
from PIL import Image
import numpy as np

img = Image.open("/vercel/share/v0-project/scripts/logo-original.jpeg").convert("RGBA")
data = np.array(img)

# The background is light gray (~#f2f2f2). Make pixels that are close to white/light-gray transparent.
# Check if R, G, B are all above a threshold (light gray/white)
r, g, b, a = data[:, :, 0], data[:, :, 1], data[:, :, 2], data[:, :, 3]

# Background pixels: all channels above 200 and similar to each other (grayish)
threshold = 210
is_bg = (r > threshold) & (g > threshold) & (b > threshold)

# Set alpha to 0 for background pixels
data[is_bg, 3] = 0

# Anti-alias: for pixels near the boundary, blend alpha based on brightness
near_threshold = 190
is_near = (r > near_threshold) & (g > near_threshold) & (b > near_threshold) & ~is_bg
brightness = (r[is_near].astype(float) + g[is_near].astype(float) + b[is_near].astype(float)) / 3
alpha_vals = ((threshold - brightness) / (threshold - near_threshold) * 255).clip(0, 255).astype(np.uint8)
data[is_near, 3] = alpha_vals

result = Image.fromarray(data)

# Crop to content (non-transparent bounding box)
bbox = result.getbbox()
if bbox:
    result = result.crop(bbox)

result.save("/vercel/share/v0-project/public/images/logo.png", "PNG")
print("Logo saved with transparent background")
print(f"Final size: {result.size}")
